crates/block-producer/src/lib.rs

```diff
@@ -42,3 +44,14 @@ const SERVER_BUILD_BATCH_FREQUENCY: Duration = Duration::from_secs(2);
 
 /// Maximum number of batches per block
 const SERVER_MAX_BATCHES_PER_BLOCK: usize = 4;
+
+// MAIN FUNCTION
+// =================================================================================================
+
+pub async fn start_block_producer(config_filepath: &Path) -> Result<()> {
```

Comment:
> Could you explain where in the binary you would add such functions?

I think, `commands/start` would be a good place for them.

> Also to note that the faucet is not a Tonic server but an Actix web server.

Not a problem, it can return Actix's status messages. The idea is to keep flexible errors in the binary, but return error codes from the library.

Reply:
Understood. Changed.